Milestone 2

okikiola11 opened the pull request:
- Prepared the directory structure
-- created a components directory in the src folder and added 4 component files to it

- Implemented the App component
-- app should render the Display and ButtonPanel component. 

- Implemented the Display component
-- It accepts the result of the calculation as an incoming prop.
-- added propTypes and DefaultProp to the Display component

- Implemented the Button component
-- It accepts a button name as a prop.
-- added propTypes and DefaultProp to the Button component

- Implemented the ButtonPanel component
-- It renders the calculator panel with all the different buttons
-- used divs to divide buttons into groups

okikiola11 committed:
Implement the App component
App render the display and ButtonPanel component
Implement the Display component
Wrap children component with React fragment

okikiola11 committed:
Remove commented codes from the index.html file

okikiola11 committed:
Implement Display component
Accept the result of the calculation as an incoming prop
Add propTypes and DefaultProps to the Display Component
Implement the Button component
Accept a button name as a prop
Add propTypes and DefaultProps to the Button Component

okikiola11 committed:
Implement the ButtonPanel component
Render the calculator panel with all the buttons
Use divs to divide buttons into groups

okikiola11 committed:
Update the text in displayCalc.js

bellom commented on package.json:
```diff
@@ -6,6 +6,7 @@
     "@testing-library/jest-dom": "^5.12.0",
     "@testing-library/react": "^11.2.6",
     "@testing-library/user-event": "^12.8.3",
+    "prop-types": "^15.7.2",
```
# [Optional]
Let's try to add `big.js` as it was stated in milestone 1
![image](https://user-images.githubusercontent.com/31897434/118617407-83533300-b7ba-11eb-96ec-73abca8e677f.png)

I made this optional because it was supposed to be a correction in milestone 1.

bellom commented on src/components/App.js:
```diff
@@ -0,0 +1,12 @@
+import ButtonPanel from './ButtonPanel';
+
+const App = () => (
+  <div className="App">
+    <h2>My React Calculator!!!</h2>
+    <DisplayCalculator />
+    <ButtonPanel />
+  </div>
```
You need to wrap both `DisplayCalculator` and `ButtonPanel` with [React.Fragment](https://reactjs.org/docs/fragments.html#short-syntax)
Reason:
We don't want to create an extra element, that's why you need to use fragment, so we are able to render without having to create an extra node wrapping `DisplayCalculator` and `ButtonPanel`.

![image](https://user-images.githubusercontent.com/31897434/118618136-391e8180-b7bb-11eb-9a16-4fe755ce9f2c.png)

bellom commented on src/components/ButtonPanel.js:
```diff
@@ -0,0 +1,39 @@
+const ButtonPanel = () => (
+  <main>
+    <div>
+      <button name="symbolReset" type="submit"> AC </button>
```
We are not going to use a button tag here, you already created a component with the button tag which is `Button`, so import `Button` from `Button.js` and use it.

And also, you need to divide buttons into groups;
![image](https://user-images.githubusercontent.com/31897434/118619881-e5149c80-b7bc-11eb-9961-a24332d8a5ff.png)

bellom commented on src/components/ButtonPanel.js:
```diff
@@ -0,0 +1,39 @@
+  <main>
+    <div>
+      <button name="symbolReset" type="submit"> AC </button>
+      <button name="symbolCalc" type="submit"> +/- </button>
+      <button name="symbolModulo" type="submit"> % </button>
+      <button name="symbolDiv" type="submit"> ÷ </button>
+    </div>
+
+    <div>
+      <button name="number7" type="submit"> 7 </button>
+      <button name="number8" type="submit"> 8 </button>
+      <button name="number9" type="submit"> 9 </button>
+      <button name="symbolMul" type="submit"> X </button>
+    </div>
+
+    <div>
+      <button name="number4" type="submit"> 4 </button>
+      <button name="number5" type="submit"> 5 </button>
+      <button name="number6" type="submit"> 6 </button>
+      <button name="symbolSubt" type="submit"> - </button>
+    </div>
+
+    <div>
+      <button name="number1" type="submit"> 1 </button>
+      <button name="number2" type="submit"> 2 </button>
+      <button name="number3" type="submit"> 3 </button>
+      <button name="symbolAdd" type="submit"> + </button>
+    </div>
+
+    <div>
+      <button name="number0" type="submit"> 0 </button>
+      <button name="symbolDec" type="submit"> . </button>
+      <button name="symbolEquals" type="submit"> = </button>
+    </div>
```
### [Optional]
Your code appears to be WET-(Write Every Time), Microverse encourage DRY-(Don't Repeat Yourself) code, In other words; you repeated the `Button` component multiple times.

You can use [objects.key](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys)  and `map` to loop through the objects. `objects.keys` for the outer loop and `map` for the inner loop.

Example; 
```
const GroupButtons = {
  Group 1: ['AC', '+/-', '%', '÷'],
  Group 2: ['7', '8', '9', 'X'],
   ............
}
```

bellom reviewed (changes requested):
## Hi @okikiola11  :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

**Heads-Up:**_Make sure to read the instructions on pathwright carefully before submitting CRR, that way, you will have a high chance of getting your CRR reviewed just once without submitting it for re-review_

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

Torres-ssf reviewed (approved):
### Milestone Approved :+1: 
Hello @okikiola11 :hand: 
Awesome job on this milestone :clap: 
Good luck with the next one

Happy coding :computer: 
**Sérgio Torres**

okikiola11 committed:
Add Big.js dependency to package.js
Add React.Fragment shorthand to App.js
Remove button tags from ButtonPanel
Import the Button from Button.js as a Component intp buttonPanel
Update text in the displayCalc file

okikiola11 committed:
Refactor buttonPanel to use DRY principle
Create GroupButtons as an object
Map through buttons
